src/app/indexes/Index.js

```diff
@@ -32,10 +37,47 @@ const ensureDocumentsAreValidStructure = (documents, structure) => {
       } catch (e) {
         throw new Error(`${e.message} (document: ${JSON.stringify(document)})`);
       }
-    })
+    });
   });
 };
 
+const attemptToStoreDocuments = async (name, keyFieldName, documents, attempts = 3) => {
+  if (attempts > 0 && documents.length > 0) {
+    const response = await storeDocumentsInIndex(name, documents);
+
+    // Response code 207 indicates there were some failures in the request, some of which can be retried.
+    // https://learn.microsoft.com/en-us/rest/api/searchservice/addupdate-or-delete-documents#response
+    if (response.statusCode === 207) {
```

Comment:
Could we return at this point for Non-207 response codes thus reducing nested _if_ blocks? Not a function change, but simply for readability.

e.g. if (response.statusCode !== 207) return;
// otherwise continue with retry policy processing where possible?

Reply:
@AMcCartneyMTH I completely agree to both this and the above point, I've made that update 👍